server/src/main/java/org/opensearch/gateway/remote/RemoteManifestManager.java

```diff
@@ -100,6 +100,9 @@ RemoteClusterStateManifestInfo uploadManifest(
         boolean committed
     ) {
         synchronized (this) {
+            if (uploadedMetadataResult.uploadedIndicesRoutingDiffMetadata != null) {
+                clusterDiffManifest.setIndicesRoutingDiffPath(uploadedMetadataResult.uploadedIndicesRoutingDiffMetadata.getUploadedFilePath());
+            }
```

Comment:
Set the routing diff from the caller of this function, this setter should not be here.

Reply:
Addressed it in revision.